source/features/warn-pr-from-master.tsx

```diff
@@ -1,41 +1,43 @@
 import React from 'dom-chef';
-import {$, elementExists} from 'select-dom';
 import * as pageDetect from 'github-url-detection';
+import elementReady from 'element-ready';
 
 import features from '../feature-manager.js';
-import getDefaultBranch, {defaultBranchOfRepo} from '../github-helpers/get-default-branch.js';
 import {getRepo} from '../github-helpers/index.js';
+import {defaultBranchOfRepo} from '../github-helpers/get-default-branch.js';
 
-async function init(): Promise<false | void> {
-	let defaultBranch;
-	if (elementExists('.is-cross-repo')) {
-		const forkedRepository = getRepo($('[title^="head: "]')!.textContent)!;
-		defaultBranch = await defaultBranchOfRepo.get(forkedRepository);
-	} else {
-		defaultBranch = await getDefaultBranch();
-	}
-
-	// Expected: /user/repo/compare/master...user:master
-	if (!location.pathname.endsWith(':' + defaultBranch)) {
-		return false;
-	}
-
-	$('.js-compare-pr')!.before(
+async function init(): Promise<void> {
+	const anchor = await elementReady('.js-compare-pr');
+	anchor?.before(
 		<div className="flash flash-error my-3">
 			<strong>Note:</strong> Creating a PR from the default branch is an <a href="https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/" target="_blank" rel="noopener noreferrer">anti-pattern</a>.
 		</div>,
 	);
 }
 
+async function isCrossRepoCompareFromMaster(): Promise<boolean> {
+	const {nameWithOwner, path, name} = getRepo()!;
+	const base = getRepo(nameWithOwner)!;
```

Comment:
Why not just:

```suggestion
	const base = getRepo()!;
```

I'd still like to see a testable `parseCompareUrl` function as suggested in https://github.com/refined-github/refined-github/pull/7683#issuecomment-2288668004

It can live in `github-helpers/parse-compare-url.ts` and be tested in `github-helpers/parse-compare-url.test.ts`. 

Line 27 shows why this is needed: `.split` will fail on the URLs:

- `https://github.com/refined-github/refined-github/compare`
- `https://github.com/refined-github/refined-github/compare/main`
- `https://github.com/refined-github/refined-github/compare/main..fregante:refined-github:main` (notice the two dots, not three)

Also the function can short-circuit when the comparison is not cross-repo, like for:

- `https://github.com/refined-github/refined-github/compare/main...rollup`

```ts
type Comparison {
	head: RepositoryInfo;
	base: RepositoryInfo;
	isCrossRepo: boolean;
}
function parseCompareUrl(pathname: string): Comparison;
```

One more weird URL:

- `https://github.com/refined-github/refined-github/compare/main...refined-github:refined-github:rollup` (it mentions the repo/user, but it's not cross-repo)